File: Part 2 - Regression/Section 1 - Simple Linear Regression/Simple_Linear_Regression.py
#Simple_Linear_Regression

#Importing_Libraries
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

#Impoting_dataset
dataset=pd.read_csv('Salary_Data.csv')
X=dataset.iloc[:,:-1].values
y=dataset.iloc[:,1].values


from sklearn.cross_validation import train_test_split
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=1/3,random_state=0)

#Fitting Simple_Linear_Regression _to_training_set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, y_train)

#Predict_test_set_result
y_pred=regressor.predict(X_test)

#Visualising_the_trainingsetresult
plt.scatter(X_train,y_train, color = 'red')
plt.plot(X_train , regressor.predict(X_train) , color = 'blue')
plt.title('Salary Vs Experiance (Training_set)')
plt.xlabel('Years of Exp')
plt.ylabel('Salary')
plt.show()

#Visualising_the_testsetresult
plt.scatter(X_test,y_test , color='red')
plt.plot(X_train , regressor.predict(X_train) , color = 'blue')
plt.title('Salary Vs Experiance (Test_set)')
plt.xlabel('Years of Exp')
plt.ylabel('Salary')
plt.show()

File: Part 2 - Regression/Section 1 - Simple Linear Regression/simple_linear_regression.R
#Simple_Linear_Regression

#importing_Data
dataset = read.csv("Salary_Data.csv")

#Training&Test_test_separation
set.seed(123)
split = sample.split(dataset$YearsExperience , SplitRatio = 2/3)
training_set = subset(dataset , split == TRUE)
test_set = subset(dataset , split == FALSE)

#Fitting_Simple_Linear_Regression_to_Traningset

regressor = lm(formula = Salary ~ YearsExperience ,data = training_set)

#Predict_with_test_set

y_pred = predict(regressor,newdata = test_set)

#Visualising_the_result_training_set
ggplot() +
  geom_point(aes(x = training_set$YearsExperience , y=training_set$Salary),
             colour = 'red') +
  geom_line(aes(x = training_set$YearsExperience , y=predict(regressor,newdata=training_set)),
            colour = 'blue') +
  ggtitle('Salary Vs YearsExperiance (Training_set)') +
  xlab('YearsofExperiance') +
  ylab('Salary')


#Visualising_the_result_test_set
ggplot() +
  geom_point(aes(x = test_set$YearsExperience , y=test_set$Salary),
             colour = 'red') +
  geom_line(aes(x = training_set$YearsExperience , y=predict(regressor,newdata=training_set)),
            colour = 'blue') +
  ggtitle('Salary Vs YearsExperiance (Test_set)') +
  xlab('YearsofExperiance') +
  ylab('Salary')
